lib/grpc/client/adapters/mint/stream_response_process.ex

```diff
@@ -0,0 +1,156 @@
+defmodule GRPC.Client.Adapters.Mint.StreamResponseProcess do
+  use GenServer
+
+  def start_link(stream, send_headers_or_trailers?) do
+    GenServer.start_link(__MODULE__, {stream, send_headers_or_trailers?})
+  end
+
+  @doc """
+  Given a pid, builds a Stream that will consume the accumulated
+    data inside this process
+  """
+  @spec build_stream(pid()) :: Elixir.Stream.t()
+  def build_stream(pid) do
+    Elixir.Stream.unfold(pid, fn pid ->
+      case GenServer.call(pid, :get_response, :infinity) do
+        nil -> nil
+        response -> {response, pid}
+      end
+    end)
+  end
+
+  @doc """
+  Cast a message to process to inform that the stream has finished
+  """
+  @spec done(pid()) :: :ok
+  def done(pid) do
+    GenServer.cast(pid, {:consume_response, :done})
+  end
+
+  @doc """
+  Cast a message to process to consume an incoming data or trailers
+  """
+  @spec consume(pid(), :data | :trailers | :headers, binary() | Mint.Types.headers()) :: :ok
+  def consume(pid, :data, data) do
+    GenServer.cast(pid, {:consume_response, {:data, data}})
```

Comment:
I don't like these `cast` calls because they'll ignore any errors or missed messages. Shouldn't they all be either a `send` or a GenServer.call?

Reply:
all handle_cast below would change to handle_call or handle_info depending on what we choose here

Reply:
So, in this case I'm treating as a cast call, cause I'm this process to be just about perform request and receive tcp messages. process the response (event the error ones) should be done on the response process. 

Cause this way I don't block the connection process from receiving other messages and also in case of an error, I would like to propagate them to the response process and put them in the Elixir.Strem. So the caller is responsible to handle errors. 

IMO I don't see lot's of benefits on using a call here, since the error treatment is done on the other hand of the processing.

Reply:
You made a similar comment bellow. I'll reply there and mark this one as solved.